jax/_src/numpy/lax_numpy.py

```diff
@@ -4328,8 +4312,10 @@ def _arange_dynamic(
   gap = step if step_gt_0 else - step
   distance = (stop - start) if step_gt_0 else (start - stop)
   size = core.max_dim(0, distance + gap - 1) // gap
-  return (array(start, dtype=dtype) +
-          array(step, dtype=dtype) * lax.iota(dtype, size))
+  sharding = canonicalize_device_to_sharding(device)
+  iota = lax.broadcasted_iota(dtype, (size, ), 0, sharding=sharding)
```

Comment:
Does iota need to be on the sharding? The `return` below should put the result automatically on the correct sharding.

Reply:
I see your point now, so if we are ok with moving `iota` between devices when we multiply `array(step, dtype=dtype, device=device) * iota` then no, `iota` can be on a default device, no need to pass sharding arg.

Reply:
@jakevdp does the sharding bit go on `iota` or on `array` below?